roles/awsenv/tasks/sg.yml

```diff
@@ -22,13 +22,24 @@
 
     - name: Create cloudfront security group
       ec2_group:
-        name: "cloudfront-{{ opg_data.stack }}"
+        name: "cloudfront-global-ip-list-{{ opg_data.stack }}"
         description: "{{ item.description }} for {{ opg_data.stack }}"
         rules: "{{ item.ruleset | rules_from_dict() }}"
         state: present
         vpc_id: "{{ app_vpc['vpc_id'] }}"
       with_items: "{{ vpc.security_groups }}"
-      when: item.Name == 'cloudfront'
+      when: item.Name == 'cloudfront_global_ip_list'
+      register: cf_sg
+
+    - name: Create second cloudfront security group
+      ec2_group:
+        name: "cloudfront-regional-edge-ip-list-{{ opg_data.stack }}"
+        description: "{{ item.description }} for {{ opg_data.stack }}"
+        rules: "{{ item.ruleset | rules_from_dict() }}"
+        state: present
+        vpc_id: "{{ app_vpc['vpc_id'] }}"
+      with_items: "{{ vpc.security_groups }}"
+      when: item.Name == 'cloudfront_regional_edge_ip_list'
```

Comment:
Are we sure that `Item` is correct here?
could you do a debug and exit task to print the contents and check `vpc.security_groups` is correct?

Reply:
When combined with this list yes.

https://gitlab.service.opg.digital/opsforks/opg-refund-deploy/blob/OPGOPS-2915/ansible/group_vars/all/env-defaults.yml#L207